zflow_runtime/src/go/mod.rs

```diff
@@ -0,0 +1,147 @@
+        let node_name = self.name.clone();
+        let mut code = PathBuf::from(&self.base_dir);
+
+        let mut inports = self.inports.clone();
+        let mut outports = self.outports.clone();
+        if inports.is_empty() {
+            inports.insert("in".to_string(), PortOptions::default());
+        }
+
+        if outports.is_empty() {
+            outports.insert("out".to_string(), PortOptions::default());
+        }
+
+        return Ok(Component::new(ComponentOptions{
+            metadata: Some(self.metadata.clone()),
+            in_ports: HashMap::from_iter(
+                inports.clone()
+                    .iter()
+                    .map(|(key, options)| (
+                        key.clone(),
+                        InPort::new(options.clone())))
+                    .collect::<Vec<_>>(),
+            ),
+            out_ports: HashMap::from_iter(
+                outports.clone()
+                    .iter()
+                    .map(|(key, options)| (key.clone(), OutPort::new(options.clone())))
+                    .collect::<Vec<_>>(),
+            ),
+            description: self.description.clone(),
+            icon: self.icon.clone(),
+            ordered: self.ordered,
+            activate_on_input: self.activate_on_input,
+            forward_brackets: self.forward_brackets.clone(),
+            // question: why doesnt a component have a graph?
+            graph: None,
+            process: Some(Box::new(move |handle| {
+                let inputs = inports.keys().collect();
+                let handle_binding = handle.clone();
+                let mut handle_binding = handle_binding.try_lock();
+
+                // todo: in order to make sense of the implementation, this variable is renamed to process_handler
+                // to remove before final PR approval/merge/submission
+                let process_handler = handle_binding
+                    .as_mut()
+                    .map_err(|_| ProcessError(String::from("Process Handle dropped. Could not lock.")))?;
+
+                let controlled = inports.iter().filter(|(key, value) | value.control)
+                    .map(|(key, value)| key).collect::<Vec<_>>();
+                let controlled_data = controlled.iter().map(|(key)| process_handler.input().get(key.clone())).collect::<Vec<_>>();
+
+                // None can take a reference? interesting....
+                if !controlled.is_empty() && controlled_data.contains(&None) {
+                    return Ok(ProcessResult::default());
+                }
+
+                // HERE BE DRAGONS. insha allah and vibes.
+                // this is the beginning of the go runtime implementation itself
+                let mut cfg = engine::Config::default();
+                cfg.trace_parser = true;
+                cfg.trace_code = true;
+                let (go_src, path) = engine::SourceReader::local_fs(code, PathBuf::from(&self.base_dir));
```

Comment:
I am curious about what happens here, because `code` is already bound to the base directory as seen in line 77.  I am guessing the second argument of `local_fs` is to tell the Go runtime where to source the other goscript sources from

Reply:
yeah. the method takes base dir and working dir. i havent checked the source (and documentations on the code per se are.... well, "kin kini"), so its mostly to be able to shut the compiler up and commit

i'm thinking the first should be "./" and the working directory should be the directory/path of the "source"?